src/integrationTest/java/io/aiven/kafka/connect/jdbc/oracle/VerifyDeleteIT.java

```diff
@@ -0,0 +1,317 @@
+                sendFutures.add(producer.send(msg));
+                if (i < numberOfTombstoneRecords) {
+                    final ProducerRecord<String, GenericRecord> record = createTombstoneRecord(i, partition);
+                    sendFutures.add(producer.send(record));
+                }
+            }
+        }
+
+        producer.flush();
+        for (final Future<RecordMetadata> sendFuture : sendFutures) {
+            sendFuture.get();
+        }
+    }
+
+    @BeforeEach
+    public void afterEach() throws SQLException {
+        executeSqlStatement(DROP_TABLE);
+        executeSqlStatement(CREATE_TABLE);
+    }
+
+    @Test
+    public void testDeleteTombstoneRecord() throws Exception {
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        consumer.assign(Collections.singleton(new TopicPartition(TEST_TOPIC_NAME, 0)));
+
+        // Start the sink connector
+        connectRunner.createConnector(sinkConnectorConfigForDelete());
+
+        sendTestData(3);
+
+        await().atMost(Duration.ofSeconds(20)).pollInterval(Duration.ofSeconds(18))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(3);
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).column("ID")
+                            .value().isEqualTo("0")
+                            .value().isEqualTo("1")
+                            .value().isEqualTo("2");
+                });
+
+        // Send test data to Kafka topic (including a tombstone record)
+        sendTestDataWithTombstone(1);
+
+        assertNotEmptyPoll(Duration.ofSeconds(50));
+
+        await().atMost(Duration.ofSeconds(20)).pollInterval(Duration.ofSeconds(18))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(2);
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).column("ID")
+                            .value().isEqualTo("1")
+                            .value().isEqualTo("2");
+                });
+    }
+
+    @Test
+    public void testWithJustTombstoneRecordInInsertMode() throws Exception {
+        // Test logic is similar to previous tests, but with tombstone records.
+        createTopic(TEST_TOPIC_NAME, 1); // Create Kafka topic matching the table name
+        consumer.assign(Collections.singleton(new TopicPartition(TEST_TOPIC_NAME, 0)));
+
+        // Start the sink connector
+        final Map<String, String> config = sinkConnectorConfigForDelete();
+        connectRunner.createConnector(config);
+
+        sendTestDataWithTombstone(2);
+
+        await().atMost(Duration.ofSeconds(20)).pollInterval(Duration.ofSeconds(18))
+                .untilAsserted(() -> {
+                    assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(0);
+                });
```

Comment:
This doesn't necessarily prove what we need it to. It's possible that we send the data, then make this assertion before the connector has had a chance to read it from Kafka and/or query the database.

If we were testing against a later version of Kafka Connect, we could wait for the connector to commit offsets (which can be verified via the `GET /connectors/{name}/offsets` endpoint introduced in version 3.5.0 as part of [KIP-875](https://cwiki.apache.org/confluence/display/KAFKA/KIP-875%3A+First-class+offsets+support+in+Kafka+Connect)).

But it's not worth adding all of that to this PR. Perhaps we could use `await().atLeast` instead of `await().atMost`, as a small temporary workaround?

Reply:
This is still unaddressed. Let me know if you have any questions!

Reply:
I tried this, but when converting it to atLeast I faced an issue where the condition is fulfilled before the duration specified in the atLeast method. I don’t think it’s possible to know the exact duration for the test to fulfill, as it depends on the time it takes for the container to come up (which depends on the available resources) and start sending messages. Please correct me if my understanding is wrong.

Reply:
Ah whoops, got the API wrong. Didn't realize it'd throw an exception if the condition were met before the `atLeast` duration expired.

As far as knowing the exact duration to wait goes--yes, you're correct that it's difficult to know exactly how long to wait. And with tombstone-only cases, if the connector is implemented correctly, then the table will have zero rows throughout the entire test.

This is why I brought up checking for committed offsets as a way to verify that the connector had consumed records from Kafka and sent them to the database (via either an insert or a delete query), but again, since it's probably not worth the work it'd take to implement that logic here, we can just wait a fixed interval and then verify that the table still has zero rows.

I'm sure there's a way to do this with Awaitility, though honestly at this point I don't know if that's really necessary. We could abandon Awaitility altogether and do something like this for the tombstone-only cases:
```java
// TODO: Instead of sleeping for a fixed interval, wait for the connector to commit offsets for the records we sent
Thread.sleep(5_000); // Give the connector at least five seconds to read our tombstone messages from Kafka
assertThat(new Table(getDatasource(), TEST_TOPIC_NAME)).hasNumberOfRows(0);
```

Reply:
Thanks for the clarification and suggestions